assets/src/edit-story/components/library/mediaLibrary.js

```diff
@@ -30,82 +44,284 @@ const Title = styled.h3`
-			rotationAngle: 0,
-		} );
+	/**
+	 * Callback of select in media picker to insert media element.
+	 *
+	 * @param {Object}attachment Attachment object from backbone media picker.
+	 */
+	const onSelect = ( attachment ) => {
+		const { url: src, mime: mimeType, width: oWidth, height: oHeight } = attachment;
+		const mediaEl = { src, mimeType, oWidth, oHeight };
+		insertMediaElement( mediaEl, DEFAULT_WIDTH );
+	};
+
+	/**
+	 * Insert element such image, video and audio into the editor.
+	 *
+	 * @param {Object}attachment Attachment object
+	 * @param {number}width      Width that element is inserted into editor.
+	 * @return {null|*}          Return onInsert or null.
+	 */
+	const insertMediaElement = ( attachment, width ) => {
+		const { src, mimeType, oWidth, oHeight } = attachment;
+		const height = getRelativeHeight( oWidth, oHeight, width );
+		const origRatio = oWidth / width;
+		if ( SUPPORTED_IMAGE_TYPES.includes( mimeType ) ) {
+			return onInsert( 'image', {
+				src,
+				width,
+				height,
+				x: 5,
+				y: 5,
+				rotationAngle: 0,
+				origRatio,
+			} );
+		} else if ( SUPPORTED_VIDEO_TYPES.includes( mimeType ) ) {
+			return onInsert( 'video', {
+				src,
+				width,
+				height,
+				x: 5,
+				y: 5,
+				rotationAngle: 0,
+				origRatio,
+				mimeType,
+			} );
+		}
+		return null;
+	};
+
+	/**
+	 * Get a formatted element for different media types.
+	 *
+	 * @param {Object}mediaEl Attachment object
+	 * @param {number}width      Width that element is inserted into editor.
+	 * @return {null|*}          Element or null if does not map to video/image.
+	 */
+	const getMediaElement = ( mediaEl, width ) => {
+		const { src, oWidth, oHeight, mimeType } = mediaEl;
+		const height = getRelativeHeight( oWidth, oHeight, width );
+		if ( SUPPORTED_IMAGE_TYPES.includes( mimeType ) ) {
+			return ( <Image
+				key={ src }
+				src={ src }
+				width={ width }
+				height={ height }
+				loading={ 'lazy' }
+				onClick={ () => insertMediaElement( mediaEl, width ) }
+			/> );
+		} else if ( SUPPORTED_VIDEO_TYPES.includes( mimeType ) ) {
+			/* eslint-disable react/jsx-closing-tag-location */
+			return ( <Video
+				key={ src }
+				width={ width }
+				height={ height }
+				onClick={ () => insertMediaElement( mediaEl, width ) }
+				onMouseEnter={ ( evt ) => {
+					evt.target.play();
+				} }
+				onMouseLeave={ ( evt ) => {
+					evt.target.pause();
+					evt.target.currentTime = 0;
+				} }
+			>
+				<source src={ src } type={ mimeType } />
+			</Video> );
+			/* eslint-enable react/jsx-closing-tag-location */
+		}
+		return null;
 	};
 
 	return (
-		<div>
+		<>
 			<Header>
 				<Title>
 					{ 'Media' }
 					{ ( ! isMediaLoaded || isMediaLoading ) &&
 						<Spinner />
 					}
 				</Title>
-				<UploadButton mediaType={ mediaType } onClose={ onClose } onSelect={ onSelect } />
+				<UploadButton
+					onClose={ onClose }
+					onSelect={ onSelect }
+				/>
 			</Header>
 
+			<SearchField>
+				<Icon icon="search"	 />
```

Comment:
^ Spacing issues here.

Reply:
Fixed in https://github.com/ampproject/amp-wp/pull/3907/commits/f9a6651ebcdd7eeaa46200cab137e53dd8fe347d